DiscordBot/DiscordCodeAlong.md

```diff
@@ -0,0 +1,280 @@
+# Discord Chat Bot
+
+In this activity, create a Discord chat bot that can respond to messages in a Discord server. How the chat bot responds and what it says is for you to decide! The chat bot could respond to common phrases, or respond to commands with useful features like rolling dice, posting pictures, or acting as a calculator.
+
+## Setting Up
+
+1. Go to [repl.it](https://repl.it)
+1. Log in
+1. Clone the [HyTechClub-Robot-Start repl](https://repl.it/@brebory/HyTechClub-Robot-Start)
+1. Make sure you have a Discord account so you can create and register your bot
+
+## Creating and Registering the Bot
+
+The first step is to navigate to the [Discord Developer Portal](https://discord.com/developers/applications). You'll be prompted to log in, so use your Discord account that you created during set up.
+
+1. Make sure that you're on the Applications tab, and then click the blue "New Application" button in the top right
+1. The "Create An Application" dialog will appear, so give your appliction a name, then click Create
+1. You will be taken to your new application's General Information page. In the left-hand sidebar, click on "Bot"
+1. On the Bot page, click the "Add Bot" button
+
+If you get an error about "Too many users share this name", then you will need to change your application's name on the General Information page first, then try again with the new name.
+
+## Saving the Bot Token in the .env File
+
+Repl.it has a special feature that allows you to create environment variables for your Repl application. In your Repl, create a new file called `.env`. In this plaintext file, you can create environment variables by making key-value pairs like this:
+
+```text
+ENVIRONMENT_VARIABLE_NAME=Value
+ANOTHER_VARIABLE=anothervalue
+```
+
+Each environment variable should be on it's own line within this file.
+
+1. In the .env file, create a new variable called `DISCORD_BOT_SECRET`
+1. Back in the Discord Developer Portal, navigate to your bot's page, and find the "Copy" button for your bot's "Token"
+![Discord Bot Settings](discord-bot-settings-cropped.jpg)
+1. Click the copy button and paste the Bot Token into your Repl's `.env` file
+
+### Code
+
+```text
+DISCORD_BOT_SECRET=ABCD.EFGHIJKLMNOPQR.STUVWXYZ
+```
+

```

Comment:
```suggestion

Your secret value should exactly match the value that you copied from your discord bot page.

```

There should be some explanation about how the secret entry above will not match what students actually have, it is important they understand it should match the token value they obtained from Discord.